### ISSUE TITLE
Devcontainer extensions not installed when using --platform on the image

### ISSUE THREAD

**Issue by xenoliss** [microsoft/vscode-remote-release#9616]
- VSCode Version: 1.87.0
- Local OS Version: MacOs
- Remote Extension/Connection Type: Containers

- Logs:

Steps to Reproduce:

**1. Create a Dockerfile with a --platform that does NOT match with your host architecture**
<img width="594" alt="image" src="https://github.com/microsoft/vscode-remote-release/assets/33523487/380d8370-5964-403f-b5f4-4e306222556a">

**2. Reference you Dockerfile in your devcontainer.json**
**3. Add extensions to the devcontainer.json**
<img width="366" alt="image" src="https://github.com/microsoft/vscode-remote-release/assets/33523487/23be3469-a6ce-4422-865e-7896ce2cd2bd">

**4. Rebuild container**
**5. Notice extensions are not installed and keep showing "Installing"**
<img width="317" alt="image" src="https://github.com/microsoft/vscode-remote-release/assets/33523487/2c134829-0799-4b3c-aef8-d1de7ddc71ba">

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Not sure what is meant by "all extensions are disabled"

**Comment by chrmarti** [comment]
Seeing that too, the extensions remain in "Installing" state.

**Comment by sandy081** [comment]
@chrmarti It would be great if you can share the least minimal steps to reproduce it?

**Comment by chrmarti** [comment]
@sandy081 I'm using the following files on an M1 mac:

`.devcontainer/devcontainer.json`:
```json
{
	"build": {
		"dockerfile": "Dockerfile"
	},
	"customizations": {
		"vscode": {
			"extensions": [
				"chrmarti.regex"
			]
		}
	}
}
```

`.devcontainer/Dockerfile`:
```Dockerfile
FROM --platform=linux/x86_64 mcr.microsoft.com/devcontainers/base:jammy
```

**Comment by xenoliss** [comment]
@sandy081 Is it really a duplicate? The issue appears **only when usging a different platform** and it is not "slow" is just never install at all.

**Comment by sandy081** [comment]
Yes, I can confirm it is a duplicate, because it is due to the same root cause that it got stuck at extension signature verification.

**Comment by xenoliss** [comment]
Is there an ETA for a fixed to be released?

**Comment by AndreuCodina** [comment]
> We are investigating, until then you can use following setting in your devcontainer.json
> 
> `"extensions.verifySignature": false`

It still happens with this setting.
